13.go

```diff
@@ -193,6 +197,10 @@ CurvePreferenceLoop:
 		return err
 	}
 
+	if _, err := c.writeRecord(recordTypeChangeCipherSpec, []byte{1}); err != nil {
```

Comment:
Should this be done unilaterally? Shouldn't this be cased off of whether the request id is empty or not (signaling Middlebox Compatibility Mode)

Reply:
A peer should ignore it, so other than saving a few bytes on the wire I think it is not worth it to add more logic for conditional use of CCS:
> An implementation may receive an unencrypted record of type change_cipher_spec consisting of the single byte value 0x01 at any time during the handshake and MUST simply drop it without further processing. 
> [..]
> This “compatibility mode” is not explicitly negotiated. [..] Either side can send change_cipher_spec at any time during the handshake, as they must be ignored by the peer.

Reply:
I tried to make this conditionally, but bogo tests fail. Investigating, but in meantime I'll keep it unconditional.